src/core/load_balancing/pick_first/pick_first.cc

```diff
@@ -1022,6 +1022,7 @@ void PickFirst::SubchannelList::SubchannelData::ProcessUnselectedReadyLocked() {
               p, p->latest_pending_subchannel_list_.get(),
               p->subchannel_list_.get());
     }
+    p->UnsetSelectedSubchannel();
```

Comment:
Can we add a test that covers this?  I'm thinking that we could get connected, then send a new update that has a READY subchannel, and then check the number of watchers to make sure it's 1 instead of 2.

Reply:
Added a test.

To trigger the original issue though, we need to enable health checking, so I added a way to enable that in the tests.

The only downside I see of this test is it seems to mostly be testing the policy's interactions with the mocks in the LB policy test lib, but I think the code path is still useful to exercise.